### PR TITLE
fix broken test about resizing column by readonly user

### PR DESCRIPTION
**Pull request by uladzimirdev**
### Description

Fix for a flaky test 

stress run [✅ 30/30](https://github.com/metabase/metabase/actions/runs/10847675551/job/30103229739)

[failing job](https://github.com/metabase/metabase/actions/runs/10838589676/job/30077431240?pr=47628)

### How to verify

run this spec locally

### Demo

https://github.com/user-attachments/assets/e5d93b0f-2e02-4e00-bdc4-c235df37ef42

### CODE REVIEW

File: e2e/test/scenarios/question/saved.cy.spec.js [hunk 1]
```diff
@@ -251,44 +251,32 @@ describe("scenarios > question > saved", () => {
...
+    cy.get("@headerCell")
+      .find(".react-draggable")
+      .trigger("mousedown", { which: 1 })
+      .trigger("mousemove", { clientX: 100, clientY: 0 })
+      .trigger("mouseup", { force: true });
+
+    cy.get("@originalWidth").then(originalWidth => {
+      cy.get("@headerCell").should($newCell => {
+        const newWidth = $newCell[0].getBoundingClientRect().width;
+        expect(newWidth).to.be.gt(originalWidth);
```
Comment (uladzimirdev) [inline e2e/test/scenarios/question/saved.cy.spec.js:276]: I saw it failing with `be.lt`

File: e2e/test/scenarios/question/saved.cy.spec.js [hunk 2]
```diff
@@ -251,44 +251,32 @@ describe("scenarios > question > saved", () => {
...
+      });
+
+    cy.get("@headerCell")
+      .find(".react-draggable")
+      .trigger("mousedown", { which: 1 })
+      .trigger("mousemove", { clientX: 100, clientY: 0 })
+      .trigger("mouseup", { force: true });
+
+    cy.get("@originalWidth").then(originalWidth => {
+      cy.get("@headerCell").should($newCell => {
```
Comment (uladzimirdev) [inline e2e/test/scenarios/question/saved.cy.spec.js:274]: `should` retries several times, `then` doesn't

File: e2e/test/scenarios/question/saved.cy.spec.js [hunk 3]
```diff
@@ -251,44 +251,32 @@ describe("scenarios > question > saved", () => {
...
-    },
-  );
+  it("'read-only' user should be able to resize column width (metabase#9772)", () => {
+    cy.signIn("readonly");
+    visitQuestion(ORDERS_QUESTION_ID);
+
+    cy.findAllByTestId("header-cell")
+      .contains("Tax")
+      .closest(".test-TableInteractive-headerCellData")
+      .as("headerCell")
```
Comment (nemanjaglumac) [inline e2e/test/scenarios/question/saved.cy.spec.js:261]: Let's fix this Frankenstein.

```suggestion
    cy.findAllByTestId("header-cell")
      .filter(":contains(Tax)")
      .as("headerCell")
```

You can also try something even simpler (but not sure if it's going to work):
`cy.findByLabel("Tax").as("headerCell");`

Comment (uladzimirdev) [reply e2e/test/scenarios/question/saved.cy.spec.js:261]: we need that parent as `headerCell` afaik, but I'll try your suggestion

Comment (nemanjaglumac) [reply e2e/test/scenarios/question/saved.cy.spec.js:261]: image

Comment (uladzimirdev) [reply e2e/test/scenarios/question/saved.cy.spec.js:261]: it worked, thanks!

**Review by romeovs — APPROVED** [review]

**Comment by uladzimirdev** [comment]
[updated stress test](https://github.com/metabase/metabase/actions/runs/10850412203/job/30111791499)

**Pull request merged**